tests/validation/test-class-amp-validation-error-taxonomy.php

```diff
@@ -0,0 +1,593 @@
+	}
+
+	/**
+	 * Test add_group_terms_clauses_filter.
+	 *
+	 * @covers \AMP_Validation_Error_Taxonomy::add_group_terms_clauses_filter()
+	 */
+	public function test_add_group_terms_clauses_filter() {
+		global $current_screen;
+
+		set_current_screen( 'front' );
+		$tested_filter = 'terms_clauses';
+		remove_all_filters( $tested_filter );
+		AMP_Validation_Error_Taxonomy::add_group_terms_clauses_filter();
+		$this->assertFalse( has_filter( $tested_filter ) );
+
+		// Only the first part of the conditional will be true, so this shouldn't add the filter.
+		$current_screen->taxonomy = AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG;
+		AMP_Validation_Error_Taxonomy::add_group_terms_clauses_filter();
+		$this->assertFalse( has_filter( $tested_filter ) );
+
+		// The entire conditional should be true, and this should add the filter.
+		$_GET[ AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_STATUS_QUERY_VAR ] = 1;
+		AMP_Validation_Error_Taxonomy::add_group_terms_clauses_filter();
+		$this->assertTrue( has_filter( $tested_filter ) );
+	}
+
+	/**
+	 * Test filter_user_has_cap_for_hiding_term_list_table_checkbox.
+	 *
+	 * @covers \AMP_Validation_Error_Taxonomy::filter_user_has_cap_for_hiding_term_list_table_checkbox()
+	 */
+	public function test_filter_user_has_cap_for_hiding_term_list_table_checkbox() {
+		$initial_caps = array( 'manage_options' );
+		$this->assertEquals( $initial_caps, AMP_Validation_Error_Taxonomy::filter_user_has_cap_for_hiding_term_list_table_checkbox( $initial_caps, array(), array() ) );
+
+		$term_id_with_description = $this->factory()->term->create( array(
+			'description' => wp_json_encode( array( 'foo' => 'bar' ) ),
+		) );
+		$args                     = array( 'delete_term', null, $term_id_with_description );
+		$this->assertEquals( $initial_caps, AMP_Validation_Error_Taxonomy::filter_user_has_cap_for_hiding_term_list_table_checkbox( $initial_caps, array(), $args ) );
+
+		$term_id_no_description = $this->factory()->term->create( array(
+			'description' => wp_json_encode( array( 'foo' => 'bar' ) ),
+		) );
+		$args                   = array( 'delete_term', null, $term_id_no_description );
+		$this->assertEquals( $initial_caps, AMP_Validation_Error_Taxonomy::filter_user_has_cap_for_hiding_term_list_table_checkbox( $initial_caps, array(), $args ) );
+	}
+
+	/**
+	 * Test filter_terms_clauses_for_description_search.
+	 *
+	 * @covers \AMP_Validation_Error_Taxonomy::filter_terms_clauses_for_description_search()
+	 */
+	public function test_filter_terms_clauses_for_description_search() {
+		$initial_where = 'tt.taxonomy IN (\'amp_validation_error\') AND tt.count > 0';
+		$clauses       = array( 'where' => $initial_where );
+		$args          = array( 'search' => 'baz' );
+
+		// The condition won't be true, so it shouldn't alter the 'where' clause.
+		$this->assertEquals( $clauses, AMP_Validation_Error_Taxonomy::filter_terms_clauses_for_description_search( $clauses, array(), $args ) );
```

Comment:
The test stubs for the `AMP_Validation_Error_Taxonomy` are implemented, but this method could use another assertion, for the [method's preg_replace()](https://github.com/Automattic/amp-wp/blob/364e03b32d8cbf5052dbfa63cb33cd6082082ef5/includes/validation/class-amp-validation-error-taxonomy.php#L613).